healthchecks/src/main/java/uk/gov/justice/services/healthcheck/healthchecks/artemis/JmsDestinationsVerifier.java

```diff
@@ -0,0 +1,35 @@
+package uk.gov.justice.services.healthcheck.healthchecks.artemis;
+
+import uk.gov.justice.services.messaging.jms.DestinationProvider;
+
+import javax.inject.Inject;
+import javax.jms.JMSException;
+import javax.jms.Session;
+import java.util.List;
+
+public class JmsDestinationsVerifier {
+
+    @Inject
+    private DestinationNamesProvider destinationNamesProvider;
+
+    @Inject
+    private DestinationProvider destinationProvider;
+
+    public void verify(Session session) throws DestinationNotFoundException {
+        List<String> destinationNames = destinationNamesProvider.getDestinationNames();
+
+        try{
+            for (final String destinationName : destinationNames) {
+                verifyThatDestinationExist(session, destinationName);
+            }
+        } catch(Exception e) {
+            throw new DestinationNotFoundException(e);
```

Comment:
Better here would be to put a message in the DestinationNotFoundException - something like 'Failed to make connction to destination 'xxxxx'. That would help enourmously with debugging problems. You'd need to move the try catch - or better still make a list of failed destinations and throw the exception with the list of failed destinations

Reply:
Addressed this partially, now exception have handled to all failed destination names which gets returned as a comma separated list in the failure json response. Not entirely sure how to log all exceptions together. In the recent commit, I'm storing only the last exception, so that gets logged. Or else I can move error logging to this class from ArtemisHealthCheck class